### PR TITLE
Added time_dependent problem to tests

### PR DESCRIPTION
**Pull request by ysimillides**
added acoustic wave equation to test, should also slightly improve coverage

### CODE REVIEW

File: test/acoustic.jl [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#bcs = [BCL.pyobject,BCD.pyobject,BCT.pyobject,BCR.pyobject]
+bcs = [BCL.pyobject,BCD.pyobject,BCT.pyobject]
+
+#bc = DirichletBC(V, 0, "on_boundary")
+A, b = assemble_system(a, L, bcs)
+
+u=FEniCS.Function(V)
+
+while t <= T
+    A, b = assemble_system(a, L, bcs)
```
Comment (ChrisRackauckas) [inline test/acoustic.jl]: You have to re-assemble every step?

Comment (ysimillides) [reply test/acoustic.jl]: yes, the boundary  condition is time_dependent

Comment (ChrisRackauckas) [reply test/acoustic.jl]: I see (that's precisely why our solution splits BCs out zzz :) )

Comment (ysimillides) [reply test/acoustic.jl]: turns out so does FEniCS to a certaint extent, if you modify the way you enter the boundary conditions. I'll upload the modified code to allow this after FEniCS'18 I guess. A very basic timing estimate shows it's slightly faster as well.

Comment (ChrisRackauckas) [reply test/acoustic.jl]: yeah that's good to know, because it seems insane to re-create the stiffness matrix every time step just for the BCs.
